clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -741,61 +767,86 @@ boolean reconcile() {
                 revokedPartitions
         );
 
-        CompletableFuture<Void> revocationResult;
-        if (!revokedPartitions.isEmpty()) {
-            revocationResult = revokePartitions(revokedPartitions);
+        // Commit offsets if auto-commit enabled before reconciling a new assignment. Request will
+        // be retried until it succeeds, fails with non-retriable error, or timer expires.
+        CompletableFuture<Void> commitResult;
+
+        if (commitRequestManager.autoCommitEnabled()) {
+            // TODO: review auto commit time boundary. This will be effectively bounded by the
```

Comment:
Do we have need a ticket for this? Not sure what it means.

Reply:
I already updated this with an initial approach aligned with the rebalance timeout as the TODO suggested and removed it.